include/net/dns_sd.h

```diff
@@ -210,6 +210,13 @@ struct dns_sd_rec {
  */
 extern const char dns_sd_empty_txt[1];
 
+/**
+ * @brief Wildcard port specifier for DNS-SD
+ *
+ * @internal
+ */
+extern const uint16_t dns_sd_port_zero;
```

Comment:
Perhaps this has been discussed earlier already, but just wondering why do we need to expose these extern variables in a public header file. Better would be to place any internal stuff to a private header file that could be placed to dns directory directly.

Reply:
All dns_sd_rec are placed in an iterable struct section which requires a `uint16_t*` to facilitate ephemeral ports. It cannot be NULL atm. Rather than having several global `uint16_t` floating around solely for this purpose, I thought it best to make 1.

Reply:
> Rather than having several global `uint16_t` floating around solely for this purpose, I thought it best to make 1.

Sure, but my question was that do we need to have these global symbols exposed in public header file, could these be placed in dns-sd private header file instead.

Reply:
> > Rather than having several global `uint16_t` floating around solely for this purpose, I thought it best to make 1.
> 
> Sure, but my question was that do we need to have these global symbols exposed in public header file, could these be placed in dns-sd private header file instead.

They need to be in a public header so that API consumers can use `DNS_SD_REGISTER_SERVICE()`, `DNS_SD_REGISTER_TCP_SERVICE()`, `DNS_SD_REGISTER_UDP_SERVICE()`, etc.

Reply:
Ok, no worries, sorry for the noise.